numba/cuda/compiler.py

```diff
@@ -943,6 +787,66 @@ def __init__(self, py_func, sigs, targetoptions):
             self.compile(sigs[0])
```

Comment:
Eager compilation here is an issue when inferred typing returns something (because the compilation can only be for a device function. What should the rules be around eager compilation and whether to infer a kernel or device function?